### DISCUSSION TITLE
Something wrong with deleteMany() ?

### DISCUSSION THREAD

**Discussion by sagerio** [directus/directus#5179] [Archive]
Hi folks,

in Typescript, I'm trying to delete a bunch of entries with  `.deleteMany()`.
At first I passed an `number[]` as argument, bot got the error `Body has to be an array of primary keys`.

Then I tried with `ID[]` - which is by the way not direct importable from `@directus/sdk`, only from `@directus/sdk/dist/types`  - but got the same error.
Same with hardcoding `.deleteMany([123])`, and the length of the array even doesn't matter...

What's the problem here?
Thanks in advance!

**Comment by rijkvanzanten** [comment]
Might be a bug in the SDK! That error message indicates that the API didn't receive the IDs in the expected location. @WoLfulus is this correct?: 

https://github.com/directus/directus/blob/b629bf61d0a8f30f9f211bdca31443279192710e/packages/sdk/src/base/items.ts#L69

In `axios` land, it's `axios.delete('/abc', { data: [] })` instead of `axios.delete('/abc', [])`

**Reply by WoLfulus** [reply]
I'll take a look! But seems like it's a bug, yes

**Reply by sagerio** [reply]
Great, thanks!
